pkg/restapi/operation/operations.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package operation
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/google/uuid"
+
+	support "github.com/trustbloc/hub-kms/pkg/internal/common"
+	"github.com/trustbloc/hub-kms/pkg/keystore"
+	"github.com/trustbloc/hub-kms/pkg/restapi/models"
+)
+
+const (
+	kmsBasePath            = "/kms"
+	keystoreEndpoint       = kmsBasePath + "/keystore"
+	createKeystoreEndpoint = kmsBasePath + "/createKeystore"
+
+	keystoreIDFormat = "urn:uuid:%s"
+
+	readRequestFailure           = "Failed to read the request body: %s"
+	createKeystoreFailure        = "Failed to create a keystore: %s"
+	receivedInvalidConfiguration = "Received invalid keystore configuration: %s"
+)
+
+// Operation defines handler logic for the KMS service.
+type Operation struct {
+	keystore keystore.Provider
+}
+
+// Handler defines an HTTP handler for the API endpoint.
+type Handler interface {
+	Path() string
+	Method() string
+	Handle() http.HandlerFunc
+}
+
+// New returns a new Operation instance.
+func New(keystore keystore.Provider) *Operation {
+	return &Operation{
+		keystore: keystore,
+	}
+}
+
+// GetRESTHandlers get all API handlers available for the KMS service.
+func (o *Operation) GetRESTHandlers() []Handler {
+	return []Handler{
+		support.NewHTTPHandler(createKeystoreEndpoint, http.MethodPost, o.createKeystoreHandler),
+	}
+}
+
+func (o *Operation) createKeystoreHandler(rw http.ResponseWriter, req *http.Request) {
+	requestBody, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		writeErrorResponse(rw, http.StatusInternalServerError, fmt.Sprintf(readRequestFailure, err))
+		return
+	}
+
+	var config models.KeystoreConfiguration
```

Comment:
what are we doing with this `config` then?

Reply:
It's not done yet (too much logic for a single PR), but the idea is to store at least the `controller` value when creating the keystore (sort of keystore's metadata) - otherwise, it's not clear how to understand later who is the "owner" of that particular keystore (and it's definitely will be needed at least from the auth prospective).